### DISCUSSION TITLE
The nuts get out whatever it is

### DISCUSSION THREAD

**Discussion by PubCyBerry** [joshreve/dactyl-keyboard#33] [Q&A]
![image](https://user-images.githubusercontent.com/44991602/130259759-a877ccfe-604a-4fee-922f-ccec98d9bd8d.png)

![image](https://user-images.githubusercontent.com/44991602/130260345-4581781a-72ba-44fe-8994-de123043b0df.png)

I tested few thumb_style option but each time I render the nuts get away.
Anyone having same problem?

**Comment by joshreve** [comment]
Pull the new versions.  I found that issue recently.  It came from a runaway setting where it was applying "trackball_in_wall"  wall widths when it shouldn't have been.  If you still see it, let me know and share your config so I can verify and fix it.

**Reply by PubCyBerry** [reply]
nut problem solved.
Current issue is that nothing happens however the config is at both gen-conf.py and run_config.py.
I don't know how to pull so I cloned it again and checked the thumb style was 'trackball_cj'.
Than ran .bat file immediately, the result was the below.

![image](https://user-images.githubusercontent.com/44991602/130346751-95619c1a-a6d4-4e0a-aaf6-03fd31c54a1f.png)

I don't know this is correct modeling but it is currenly invariant under config change.
When I changed engine sold to cadquery, it worked. So it refers config file. 
But still mystery why anything below the engine is not applied.
For example, This is the result of show_pcb=false, thumb_style=MINIDOX.

![image](https://user-images.githubusercontent.com/44991602/130349213-a3b84687-9674-4715-8953-e783bb3b2acf.png)

absolutely nothing changed.

Lastly, I tried git revert or git checkout after cloning but failed. Is this normal?
![image](https://user-images.githubusercontent.com/44991602/130349270-c71fe9a0-0b63-4fba-9561-5ef133d94963.png)

**Reply by PubCyBerry** [reply]
One more about .bat file on windows 10.
DM-config is binding src and thing. 
generate_configuration.py needs ../config folder so it makes error.
I don't know this happens same on linux.

![image](https://user-images.githubusercontent.com/44991602/130349392-c547855c-1088-44bf-88d3-855e4471cac9.png)

**Reply by joshreve** [reply]
Ugh, that is my bad.  Has some new workflow changes and I didn't do sufficient testing.  Will correct today.  I will probably have to set up unit tests with the number of configs going.

**Reply by joshreve** [reply]
Fixing the batch file right now.  I'm going to look at the other one.  I haven't seen that error before.  I will try the issues you see on my test clone to see what I get.  It seems like occasionally my local changes are quite reflected properly in the repo.  If it really is damaged, I may have to rebase.

**Comment by joshreve** [comment]
Committing  now.  I fixed the bat files to add the extra directories.  You probably need to delete and recreate the apps in docker to get the changes.  I looked at the no change issue and I'm not seeing that effect.  Check the "save_dir" and "config_name" in case it is putting somewhere else.  I am seeing the changes in my runs.  I am trying to transition to getting some public configs into the repo for people to share and I will admit, I didn't quite get the workflow right.  One other thing, if you are still on the trackball branch, that may cause trouble.  I merged trackball with master.  I should make sure I bring trackball up to a stable point.

**Reply by joshreve** [reply]
I will have to take a look at everything holistically soon.  I am concerned that I am starting to support too many workflows and may need to find a good way to consolidate and minimize maintenance overhead.  I believe your specific issue should be fixed at this point and if you pull master and have issues, let me know.

**Reply by PubCyBerry** [reply]
Excellent.
.bat issue solved and dactyl_manuform refers config correctly.
Very Good.

I found another issue.
generate_config.py(or DM-config) seems not working.
When I directly change config.json , it is applied to model.
But when I give some change at gen-config.py, It doesn't modify config.json.


I suggest few things. 
You are managing both window and linux system(.bat and .sh), which make your labor double. But you already built docker image, that is independent of OS. Make some optimization wrt docker and leave docker hub image link instead.  
Lastly, I recommend binding repository folder instead of binding 3 folders individually.

This project is very useful. I hope it goes on.
Thank you for your time.

**Reply by joshreve** [reply]
I will fix the generate command in the short run.  I want to stabilize this branch and open a dev branch to redo the workflow a bit.  I let too many factors go active on the same branch so I just need to make sure people can run it and then get back to a long term fix.

 I will be looking at how to minimize the docker config aspects.  I will look at the docker hub approach.  The other thing is minimizing the OS specific content as you pointed out.  Also, good call on the binding.  There is no reason for the separate ones.  Those will all go into the dev branch, for now I just need config working and it will at least be back to fully functional.

**Comment by joshreve** [comment]
Should be fixed.  Let me know if it is still misbehaving.  Thanks for the comments on the workflow.  I'll be messing with that soon.   Hopefully I can nail it down this time to minimize long term overhead.
